### PR TITLE
feat: add richer error messages to flox push

### PR DESCRIPTION
**Pull request by ysndr**
* feat: add richer error messages to `flox push`
* feat: add `flox edit --name`

### CODE REVIEW

File: cli/flox/src/commands/environment.rs [hunk 1]
```diff
@@ -1059,22 +1102,41 @@ impl Push {
...
+        let name = &pointer.name;
 
-                Please login to floxhub with `flox auth login`
-                "})
+        let message = match err {
+            ManagedEnvironmentError::AccessDenied => formatdoc! {"
+                ❌  You do not have permission to write to {owner}/{name}
+            "}.into(),
+            ManagedEnvironmentError::Diverged if create_remote => formatdoc! {"
+                ❌  An environment named {owner}/{name} already exists!
```
Comment (mkenigs) [inline cli/flox/src/commands/environment.rs:1126]: question nonblocking: (this is contradictory to the ticket so no need to change) do we need to mention the possibility of the environment having diverged? It's somewhat implied by the `flox delete && flox pull` but not super obvious.  @ghudgins

File: cli/flox/src/commands/environment.rs [hunk 2]
```diff
@@ -1059,22 +1102,41 @@ impl Push {
...
+
+                To rename your environment: 'flox edit --name <new name>'
+                To pull and manually re-apply your changes: 'flox delete && flox pull -r {owner}/{name}'
```
Comment (mkenigs) [inline cli/flox/src/commands/environment.rs:1129]: suggestion nonblocking: make it more obvious this is destructive, or suggest a way to stash existing work

**Review by mkenigs — CHANGES_REQUESTED** [review]

File: cli/flox/src/commands/environment.rs [hunk 3]
```diff
@@ -1059,22 +1102,41 @@ impl Push {
         let path_environment =
             path_environment::PathEnvironment::open(path_pointer, dot_flox_path, &flox.temp_dir)?;
 
-        let env =
-            ManagedEnvironment::push_new(flox, path_environment, owner.parse().unwrap(), force)
-                .map_err(Self::convert_error)?;
+        let pointer = ManagedPointer::new(owner.clone(), path_environment.name(), &flox.floxhub);
+
+        let env = ManagedEnvironment::push_new(flox, path_environment, owner, force)
+            .map_err(|err| Self::convert_error(err, pointer, true))?;
```
[the remark this answers is not in the document]
Comment (ysndr) [reply cli/flox/src/commands/environment.rs:1108]: bc the path environment is already moved into push_new and thus not accessible in map_err

File: cli/tests/environment-edit.bats [hunk 4]
```diff
@@ -208,3 +208,31 @@ EOF
...
+# bats test_tags=edit:rename-remote
+@test "'flox edit --name' fails with a remote environment" {
+  floxhub_setup "owner"
+
+  "$FLOX_BIN" init --name name
+  "$FLOX_BIN" push --owner "owner"
+
+  run "$FLOX_BIN" edit --remote "owner/name" --name "renamed"
+  assert_failure
+  assert_output --partial "Can not rename a environments on floxhub"
```
[the remark this answers is not in the document]
Comment (ysndr) [reply cli/tests/environment-edit.bats:237]: Thanks for changing this too ^^

File: cli/flox/src/commands/environment.rs (hunk 2 above)
Comment (ysndr) [reply cli/flox/src/commands/environment.rs:1129]: Yeah there's more we could add to the error
I wanted to keep it as close to the AC as possible

Also prefer to merge this first and then iterate on the wording..

**Review by mkenigs — APPROVED** [review]

**Pull request merged**